docs/checklist_templates/eip_testing_checklist_template.md

```diff
@@ -0,0 +1,289 @@
+        - [ ] Calldata too short, given different inputs, where the value represents a correct but truncated input to the contract.
+        - [ ] Calldata too long, given different inputs, where the value represents a correct input to the contract with padded zeros.
+- [ ] Excessive Gas Cases
+    - [ ] If possible, simulate a scenario where the execution of the contract at the end of the block execution by the system address would result in excessive gas usage (100 million gas or more).
+    - [ ] Verify spending all block gas in calls to system contract (100 million gas or more).
+- [ ] System Contract Deployment
+    - [ ] Verify block execution behavior after fork activation if the system contract has not been deployed.
+    - [ ] Verify deployment transaction results in the system contract being deployed to the expected address.
+- [ ] Contract Variations
+    - [ ] Verify execution of the different variations of the contract for different networks (if any) results in the expected behavior
+- [ ] Contract Substitution
+    - [ ] Substitute the contract to modify its behavior when called by the system address (at the end of the block execution):
+        - [ ] Modified return value lengths
+        - [ ] Modify emitted logs
+- [ ] Fork transition
+    - [ ] Verify calling the system contract before its activation fork results in correct behavior (depends on the system contract implementation).
+
+### Framework Changes
+
+- [ ] Add system contract address to relevant methods in the fork where the EIP is introduced in `src/ethereum_test_forks/forks/forks.py`
+- [ ] Add system contract bytecode to the returned value of `pre_allocation_blockchain` in the fork where the EIP is introduced in `src/ethereum_test_forks/forks/forks.py`
+
+## New Transaction Type
+
+### Test Vectors
+
+- [ ] Intrinsic Gas Costs
+    - [ ] Transaction validity: For each new field that affects the intrinsic gas cost of the transaction:
+        - [ ] Verify the transaction (and the block it is included in) is valid by providing the exact intrinsic gas as `gas_limit` value to the transaction with all multiple combinations of values to the field.
+        - [ ] Verify the transaction (and the block it is included in) is invalid by providing the exact intrinsic gas minus one as `gas_limit` value to the transaction with all multiple combinations of values to the field.
+- [ ] Encoding Tests (RLP, SSZ)
+    - [ ] Verify correct transaction rejection due to incorrect field sizes
+    - [ ] Verify correct transaction rejection if the fields particular to the new transaction types are missing
+    - [ ] Verify correct transaction rejection if the transaction type contains extra fields
+    - [ ] If the transaction contains fields with new serializable types, perform all previous tests on the new type/field
+- [ ] RPC Tests
+    - [ ] * Verify `eth_estimateGas` behavior for different valid combinations of the new transaction type
```

Comment:
How to test this? This depends upon the parent block it is being called on, and it will also depend upon what params are used in the block we are adding this tx (for instance: timestamp, could have a skipped block so this block is not T+12 but T+24)

Reply:
I think https://github.com/ethereum/execution-apis has the correct setup to make this more deterministic.

In this repository we could try to use `execute` to make these tests deterministic.